identity/identity.go

```diff
@@ -30,11 +28,50 @@ func NewIdentityClient(address string, s signer.DynamicSigner) (IdentityClient,
 	return &identityClient{auth: auth, s: s, cm: crypto.NewCryptoManager()}, nil
 }
 
-func (c *identityClient) Register(params params.RegisterIdentityParams) error {
-	return errors.New("not implemented")
+func (c *identityClient) Register(p *params.RegisterIdentityParams) error {
+	did := c.s.GetDid()
+	if p.Challenge != nil {
+		nonce := p.Challenge.Nonce
+		signed, err := c.s.Sign(signer.Authentication, []byte(did+"."+nonce))
+		if err != nil {
+			return err
+		}
+
+		err = c.auth.AuthnWithDidRegister(did, nonce, signed)
+		if err != nil {
```

Comment:
`if err := c.auth.AuthnWithDidRegister(did, nonce, signed); err != nil {`

Reply:
Sounds good

Reply:
I've refactored the lines to follow this form